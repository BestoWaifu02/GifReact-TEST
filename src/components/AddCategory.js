import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log("hanlde submit", inputValue);
    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
/* Cuando se envia un form este hace que la pagina se recargue por completo lo cual es malo porque react no recarga la pagina sino componentes, lo que se hace en este caso es quitar el evento de submit llamando la funcion handleSubmit

 Nombre del componentes, al setCategories lo hace obligatorio :) */
