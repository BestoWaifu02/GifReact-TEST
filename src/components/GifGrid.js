import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn"> {category} </h3>

      {loading && <p className="animate__animated animate__flash">Loading</p>}

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
/*MI ENTENDIMIENTO 
    Primero hacemos la peticion fetch con el async y el await, obnetemos los datos de la API y los filtramos para que solo traiga el id,title,url pero esto se saca de la data para despues ser mapeada y el array que te regresa se pone con la const gif la cual se va al setImages del hook y de ahi a la const de images para despues sacar sus propiedades  
  const [images, setImages] = useState([]);
  Este hook hace que no se vuelva a ejecutar la peticion porque tenemos un boton que agrega al contador un numero y por eso recarga el componente pero si usamos este hook la peticion solo se hara 1 vez
  useEffect(() => {
    importamos el getGifts y como retorna una promesa usamos el then 
    getGifts(category).then(setImages)
  }, [category]); 
  cambio de nombre*/

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
