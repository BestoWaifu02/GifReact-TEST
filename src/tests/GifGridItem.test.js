import { shallow } from "enzyme";
import { GifGridItem } from "../components/GifGridItem";
import React from "react";

import "@testing-library/jest-dom/extend-expect";
describe("Pruebas en el componente GifGridItem", () => {
  //Props
  const title = "Example";
  const url = "https://";

  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  test("Debe hacer match con la snapshot del componente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener un Parrafo con el title", () => {
    //Obtenemos el parrafo
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("Debe de tener la imagen igual al de la URL y alt de los", () => {
    const img = wrapper.find("img");
    //Ver props html de la IMG
    /* console.log(img.html())
    console.log(img.prop('src')) */
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Que el div tenga sus clases (buscando solo 1)", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");
    // console.log(div.html());
    //Vemos que className incluya animate_fadeIn y que sea true
    expect(className.includes("animate__animated animate__fadeIn")).toBe(true);
  });
});
