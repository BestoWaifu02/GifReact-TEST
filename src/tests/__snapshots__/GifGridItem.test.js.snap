// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem Debe hacer match con la snapshot del componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Example"
    src="https://"
  />
  <p>
     
    Example
     
  </p>
</div>
`;
